### PR TITLE
Whaon patch issue 802

### PR DESCRIPTION
**Pull request by whaon**
fix enhancement issue #802

### CODE REVIEW

File: core/src/main/java/org/openstack4j/openstack/internal/BaseOpenStackService.java [hunk 1]
```diff
@@ -33,6 +33,15 @@
 
     ServiceType serviceType = ServiceType.IDENTITY;
     Function<String, String> endpointFunc;
+    
+    private static ThreadLocal<String> reqIdContainer = new ThreadLocal<String>();
+    
+    public static final String X_OPENSTACK_REQUEST_ID = "x-openstack-request-id";
```
Comment (vinodborole) [inline core/src/main/java/org/openstack4j/openstack/internal/BaseOpenStackService.java]: Can these be part of https://github.com/ContainX/openstack4j/blob/master/core/src/main/java/org/openstack4j/core/transport/ClientConstants.java class?

**Comment by vinodborole** [comment]
@auhlig  Can you please review?

**Comment by auhlig** [comment]
Besides the comment above LGTM

File: core/src/main/java/org/openstack4j/openstack/internal/BaseOpenStackService.java (hunk 1 above)
Comment (whaon) [inline core/src/main/java/org/openstack4j/openstack/internal/BaseOpenStackService.java]: it's a good idea

**Comment by whaon** [comment]
Hi,it's ok in jdk7 but failed in jdk8.
I can't get the job log,pls give me a hand
Thanks

**Comment by auhlig** [comment]
Connection reset while trying to get a dependency from maven central -.-
Could you close and reopen the PR to trigger travis build again?

**Comment by auhlig** [comment]
Good to merge @vinodborole ?

**Comment by vinodborole** [comment]
LGTM

**Pull request merged**